File: apps/api/BHMHockey.Api/Models/Entities/TournamentAnnouncement.cs
namespace BHMHockey.Api.Models.Entities;

public class TournamentAnnouncement
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid TournamentId { get; set; }
    public Tournament Tournament { get; set; } = null!;

    public string Title { get; set; } = null!;
    public string Body { get; set; } = null!;

    // Target: "All", "Captains", "Admins" (or null when TargetTeamIds is set for specific teams)
    public string? Target { get; set; } = "All";

    // JSONB array of team Guids - when set, announcement targets specific teams
    public string? TargetTeamIds { get; set; }

    public Guid CreatedByUserId { get; set; }
    public User CreatedByUser { get; set; } = null!;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }  // Soft delete
}

File: packages/shared/src/types/tournamentAnnouncement.ts
// ============================================
// Tournament Announcement Types
// ============================================

// Target options for announcements
export type AnnouncementTarget = 'All' | 'Captains' | 'Admins';

// Response DTO
export interface TournamentAnnouncementDto {
  id: string;
  tournamentId: string;
  title: string;
  body: string;
  target: AnnouncementTarget | null;  // null when targeting specific teams
  targetTeamIds: string[] | null;     // Team IDs when targeting specific teams
  createdByUserId: string;
  createdByFirstName: string;
  createdByLastName: string;
  createdAt: string;  // ISO date string
  updatedAt: string | null;
}

// Create request
export interface CreateTournamentAnnouncementRequest {
  title: string;
  body: string;
  target?: AnnouncementTarget | null;
  targetTeamIds?: string[] | null;
}

// Update request
export interface UpdateTournamentAnnouncementRequest {
  title?: string;
  body?: string;
  target?: AnnouncementTarget | null;
  targetTeamIds?: string[] | null;
}
